source/libip6string/stoip6.c

```diff
@@ -70,14 +70,39 @@ void stoip6(const char *ip6addr, size_t len, void *dest)
         memset(addr, 0, 16 - field_no * 2);
     }
 }
-unsigned char  sipv6_prefixlength(const char *ip6addr)
+
+unsigned char sipv6_prefixlength(const char *ip6addr)
 {
     char *ptr = strchr(ip6addr, '/');
     if (ptr) {
         return (unsigned char)strtoul(ptr + 1, 0, 10);
     }
     return 0;
 }
+
+int stoip6_prefix(const char *ip6addr, void *dest, int16_t *prefix_len_out)
+{
+    size_t addr_len;
+
+    if (prefix_len_out) {
+        *prefix_len_out = -1;
+    }
+
+    addr_len = strlen(ip6addr);
+    char *ptr = strchr(ip6addr, '/');
+    if (ptr) {
+        addr_len = ptr - ip6addr;
+        if (prefix_len_out) {
+            *prefix_len_out = strtoul(ptr + 1, 0, 10);
```

Comment:
Maybe be a little bit stricter here, in the way that prefixlength isn't? Check that we consumed the entire remaining string, and it's in the range 0-128?